From 6db536746a28ddf67cb04eb69b42fe411af587a9 Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Wed, 31 Oct 2018 23:13:29 +0530
Subject: [PATCH 0129/1851] ASoC: xlnx: disable strict check of AES parameters

commit 575a7a06d588895495c56930439e86197b6f52a2 from
https://github.com/Xilinx/linux-xlnx.git

AES audio stream parameters is observed to be not always available
to compare against user provided parameters during capture. So, disable
comparing and just provide AES stream parameters as information to user.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Sandip Kothari <sandipk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index 48e584b0ec17..a0637b58f036 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -516,19 +516,9 @@ static int xlnx_formatter_pcm_hw_params(struct snd_pcm_substream *substream,
 				status = parse_consumer_format(aes_reg1_val,
 							       aes_reg2_val,
 							       aes_params);
-
-			if (status) {
-				kfree(aes_params);
-				return status;
-			}
-
-			if (active_ch != aes_params->channels ||
-			    bits_per_sample != aes_params->sig_bits ||
-			    sample_rate != aes_params->srate) {
-				dev_warn(component->dev, "capture parameters mismatch!\n");
-				kfree(aes_params);
-				return -EINVAL;
-			}
+			dev_info(component->dev, "rate = %d bit depth = %d ch = %d\n",
+				 aes_params->srate, aes_params->sig_bits,
+				 aes_params->channels);
 			kfree(aes_params);
 		}
 	}
-- 
2.31.1

